js/state/siteUtil.js

```diff
@@ -417,6 +417,10 @@ module.exports.removeSite = function (state, siteDetail, tag, reorder = true, sy
     site = site.set('tags', tags)
     return state.setIn(stateKey, site)
   } else {
+    if (!tag && module.exports.isHistoryEntry(siteDetail)) {
+      // Delete the site from history
+      return state.deleteIn(stateKey)
```

Comment:
but I tried this PR with STR in #7238. bookmark will disappear and pinned tab will also disappear after relaunch